src/sqlfluff/rules/L064.py

```diff
@@ -0,0 +1,234 @@
+                        [
+                            CodeSegment(
+                                raw=fixed_string,
+                                name="quoted_literal",
+                                type="literal",
+                            )
+                        ],
+                    )
+                ],
+                description=(
+                    "Inconsistent use of preferred quote style '"
+                    f"{QUOTES_MAPPING[self.preferred_string_quotes]['common_name']}'. "
+                    f"Use {fixed_string} instead of {context.segment.raw}."
+                ),
+            )
+
+        return None
+
+    def _sub_twice(self, regex: regex.Pattern, replacement: str, original: str) -> str:
+        """Replace `regex` with `replacement` twice on `original`.
+
+        This is used by string normalization to perform replaces on
+        overlapping matches.
+        """
+        return regex.sub(replacement, regex.sub(replacement, original))
+
+    def _normalize_preferred_string_quotes(
+        self, s: str, preferred_quote_char: str, alternate_quote_char: str
+    ) -> str:
+        """Prefer double quotes but only if it doesn't cause more escaping.
+
+        Adds or removes backslashes as appropriate. Doesn't parse and fix
+        strings nested in f-strings.
+        """
+        value = s.lstrip(STRING_PREFIX_CHARS)
+
+        if value[:3] == preferred_quote_char * 3:
+            # TODO: Are we not replacing unnecessary quotes here?
+            return s
+        elif value[0] == preferred_quote_char:
+            # quotes are alright already,
+            # but maybe we can remove some unneeded escapes or reduce the number
+            # of escape using alternate_quote_char ?
+            orig_quote = preferred_quote_char
+            new_quote = alternate_quote_char
+        elif value[:3] == alternate_quote_char * 3:
+            orig_quote = alternate_quote_char * 3
+            new_quote = preferred_quote_char * 3
+        elif value[0] == alternate_quote_char:
+            orig_quote = alternate_quote_char
+            new_quote = preferred_quote_char
+        else:
+            self.logger.debug(
+                "Found quoted string %s using neither preferred quote char %s "
+                "nor alternate_quote_char %s. Skipping...",
+                s,
+                preferred_quote_char,
+                alternate_quote_char,
+            )
+            return s
+
+        first_quote_pos = s.find(orig_quote)
+        if first_quote_pos == -1:
+            return s  # There's an internal error
+
+        prefix = s[:first_quote_pos]
+        unescaped_new_quote = regex.compile(rf"(([^\\]|^)(\\\\)*){new_quote}")
+        escaped_new_quote = regex.compile(rf"([^\\]|^)\\((?:\\\\)*){new_quote}")
+        escaped_orig_quote = regex.compile(rf"([^\\]|^)\\((?:\\\\)*){orig_quote}")
+        body = s[first_quote_pos + len(orig_quote) : -len(orig_quote)]
+
+        if "r" in prefix.casefold():
```

Comment:
Is there an advantage to using `casefold` here, given that `r` is a simple alphabetical character, not a fancy Unicode character?

Reply:
It's really just used for case sensitivity here. I.e. the raw prefix could be `r` or `R`. Let me just use `lower` 👍